src/github.rs

```diff
@@ -0,0 +1,272 @@
+use crate::install::AssetType;
+use color_eyre::{eyre::eyre, Result};
+use reqwest::Client;
+use serde_json::Value;
+
+pub struct GithubReleaseRepository {
+    pub base_url: String,
+    pub organisation_name: String,
+    pub repository_name: String,
+}
+
+impl GithubReleaseRepository {
+    pub fn new(base_url: &str, organisation_name: &str, repository_name: &str) -> Self {
+        Self {
+            base_url: base_url.to_string(),
+            organisation_name: organisation_name.to_string(),
+            repository_name: repository_name.to_string(),
+        }
+    }
+
+    /// Retrieves the asset name of the latest release for the specified type and platform.
+    ///
+    /// # Arguments
+    ///
+    /// * `asset_type` - Either the client or node.
+    /// * `platform` - The target triple platform of the binary to be installed.
+    ///
+    /// # Returns
+    ///
+    /// A tuple containing the name of the asset and the version number for the latest release.
+    ///
+    /// # Errors
+    ///
+    /// Returns an `eyre::Report` if an error occurs during the retrieval process. Possible error
+    /// conditions include network errors, JSON parsing errors, and missing or invalid data in the
+    /// response body.
+    pub async fn get_latest_asset_name(
+        &self,
+        asset_type: AssetType,
+        platform: &str,
+    ) -> Result<(String, String)> {
+        let client = Client::new();
+        let response = client
+            .get(format!(
+                "{}/repos/{}/{}/releases/latest",
+                self.base_url, self.organisation_name, self.repository_name
+            ))
+            .header("User-Agent", "safeup")
+            .send()
+            .await?;
+        let body = response.text().await?;
```

Comment:
I would perhaps insert an empty line between every ~3 or so variable assignments, for readability (depends on how many lines a single assignment takes up as well).